src/MongoDbEventStoreAdapter.php

```diff
@@ -0,0 +1,218 @@
+                    $stream->streamName()->toString(),
+                    __CLASS__
+                )
+            );
+        }
+
+        $this->appendTo($stream->streamName(), $stream->streamEvents());
+    }
+
+    /**
+     * @param StreamName $streamName
+     * @param DomainEvent[] $streamEvents
+     * @throws \Prooph\EventStore\Exception\StreamNotFoundException If stream does not exist
+     * @return void
+     */
+    public function appendTo(StreamName $streamName, array $streamEvents)
+    {
+        foreach ($streamEvents as $event) {
+            $this->insertEvent($streamName, $event);
+        }
+    }
+
+    /**
+     * @param StreamName $streamName
+     * @param null|int $minVersion
+     * @return Stream|null
+     */
+    public function load(StreamName $streamName, $minVersion = null)
+    {
+        $events = $this->loadEventsByMetadataFrom($streamName, array(), $minVersion);
+
+        return new Stream($streamName, $events);
+    }
+
+    /**
+     * @param StreamName $streamName
+     * @param array $metadata
+     * @param null|int $minVersion
+     * @return DomainEvent[]
+     */
+    public function loadEventsByMetadataFrom(StreamName $streamName, array $metadata, $minVersion = null)
+    {
+        $collection = $this->mongoClient->selectCollection($this->dbName, $this->getCollection($streamName));
+
+        if (null !== $minVersion) {
+            $metadata['version'] = ['$gt' => [$minVersion]];
+        }
+
+        $results = $collection->find($metadata);
+
+        foreach ($results as $eventData) {
+            $payload = Serializer::unserialize($eventData['payload'], $this->serializerAdapter);
+
+            $eventClass = $eventData['event_class'];
+
+            //Add metadata stored in table
+            foreach ($eventData as $key => $value) {
+                if (! in_array($key, $this->standardColumns)) {
+                    $metadata[$key] = $value;
+                }
+            }
+
+            $events[] = $eventClass::fromArray(
+                [
+                    'uuid' => $eventData['event_id'],
+                    'name' => $eventData['event_name'],
+                    'version' => (int)$eventData['version'],
+                    'created_at' => $eventData['created_at'],
+                    'payload' => $payload,
+                    'metadata' => $metadata
+                ]
+            );
+        }
+
+        return $events;
+    }
+
+    /**
+     * Insert an event
+     *
+     * @param StreamName $streamName
+     * @param DomainEvent $e
+     * @return void
+     */
+    protected function insertEvent(StreamName $streamName, DomainEvent $e)
+    {
+        $eventData = array(
+            'event_id' => $e->uuid()->toString(),
+            'version' => $e->version(),
+            'event_name' => $e->messageName(),
+            'event_class' => get_class($e),
+            'payload' => Serializer::serialize($e->payload(), $this->serializerAdapter),
```

Comment:
Do we need the serializer for mongoDb? DomainEvent::payload must only contain scalar types and arrays so we can persist the payload as is. This would allow querying events with specific payload. Something that is not supported by the event store but custom "event replay scripts" may want to make use of such a feature.